src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs

```diff
@@ -863,14 +863,26 @@ public async Task CollectionConstructorsKnownToRequireComparer_DiagnosticAsync()
                         @"
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.CodeAnalysis;
+
 public class C
 {
-    public void MethodWithDiagnostics()
+    public void MethodWithDiagnostics(IEnumerable<ISymbol> symbols)
     {
+        var kvps = symbols.Select(s => new KeyValuePair<ISymbol, int>(s, 0));
+
         [|new Dictionary<ISymbol, int>()|];
+        [|new Dictionary<ISymbol, int>(42)|];
+        [|new Dictionary<ISymbol, int>(kvps)|];
+
         [|new HashSet<ISymbol>()|];
+        [|new HashSet<ISymbol>(42)|];
+        [|new HashSet<ISymbol>(symbols)|];
+
         [|new ConcurrentDictionary<ISymbol, int>()|];
+        [|new ConcurrentDictionary<ISymbol, int>(kvps)|];
+        [|new ConcurrentDictionary<ISymbol, int>(1, 42)|];
```

Comment:
Can we add unit tests where existing invocations and object creations use named arguments? Also do you know if we have existing tests where code already passes a comparer and hence no diagnostic is raised?

Reply:
I have added one test with named argument and one test with passing ` SymbolEqualityComparer.Default`.